test/iaas.AzureClient.spec.js

```diff
@@ -0,0 +1,366 @@
+                <Properties>
+                  <Last-Modified>Wed, 09 Sep 2016 09:20:02 GMT</Last-Modified>  
+                  <Etag>0x8CBFF45D8A29A19</Etag>  
+                  <Content-Length>100</Content-Length>  
+                  <Content-Type>text/html</Content-Type>  
+                  <Content-Encoding />  
+                  <Content-Language>en-US</Content-Language>  
+                  <Content-MD5 />  
+                  <Cache-Control>no-cache</Cache-Control>  
+                  <BlobType>BlockBlob</BlobType>  
+                  <LeaseStatus>unlocked</LeaseStatus>  
+                </Properties>
+                <Metadata>  
+                  <Color>blue</Color>  
+                  <BlobNumber>01</BlobNumber>  
+                </Metadata>  
+              </Blob>
+              <Blob>
+                <Name>${blobList[1]}</Name>
+                <Url>https://${settings.storageAccount}.blob.core.windows.net/${settings.container}/${blobList[1]}</Url>  
+                <Properties>  
+                  <Last-Modified>Wed, 09 Sep 2016 09:20:02 GMT</Last-Modified>  
+                  <Etag>0x8CBFF45D8B4C212</Etag>  
+                  <Content-Length>5000</Content-Length>  
+                  <Content-Type>application/octet-stream</Content-Type>  
+                  <Content-Encoding>gzip</Content-Encoding>  
+                  <Content-Language />  
+                  <Content-MD5 />  
+                  <Cache-Control />  
+                  <BlobType>BlockBlob</BlobType>  
+                </Properties>  
+                <Metadata>  
+                  <Color>green</Color>  
+                  <BlobNumber>02</BlobNumber>  
+                </Metadata>  
+              </Blob>
+            </Blobs>  
+            <NextMarker />   
+          </EnumerationResults>`
+        };
+        mocks.azureClient.list(settings.container, undefined, expectedResponse);
+        return client.list()
+          .then(files => {
+            expect(files).to.have.length(blobList.length);
+            expect(files[0].name).to.eql(blobList[0]);
+            expect(files[1].name).to.eql(blobList[1]);
+            expect(files[0].isTruncated).to.eql(false);
+            expect(files[1].isTruncated).to.eql(false);
+            mocks.verify();
+          });
+      });
+
+      it('blob upload should be successful', function () {
+        mocks.azureClient.upload(`/${settings.container}/${blobName}`, {
+          status: 201,
+          headers: {
+            'transfer-encoding': 'chunked',
+            'content-md5': '+v4bYMJBB8zY9FYiE+RISQ==',
+            'last-modified': 'Thu, 05 Jan 2017 03:13:04 GMT',
+            etag: '"0x8D43518C381D87C"',
+            server: 'Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0',
+            'x-ms-request-id': '5dc67ddb-0001-003d-7701-672321000000',
+            'x-ms-version': '2016-05-31',
+            'x-ms-request-server-encrypted': 'false',
+            date: new Date().toISOString(),
```

Comment:
Do you have to mock it with all these headers explicitly? can you not just mock with URL? Reason being if we upgrade the azure library and if the lib starts sending in new headers then these tests would also fail we might have to fix for errors which are non-functional. So please verify if these mocks can be generic. 
Also in the above i see date is being set will this match the date sent at request time? for matching dates always use sinon faketimers in tests.

Reply:
These are the response headers from Azure and independent of library.

Reply:
Though removed many headers.